Gems/Terrain/Code/Tests/TerrainSystemTest.cpp

```diff
@@ -902,4 +902,168 @@ namespace UnitTest
+        const AZ::Vector2 stepSize(1.0f);
+
+        const SurfaceData::SurfaceTag tag1 = SurfaceData::SurfaceTag("tag1");
+        const SurfaceData::SurfaceTag tag2 = SurfaceData::SurfaceTag("tag2");
+        const SurfaceData::SurfaceTag tag3 = SurfaceData::SurfaceTag("tag3");
+
+        AzFramework::SurfaceData::SurfaceTagWeight tagWeight1;
+        tagWeight1.m_surfaceType = tag1;
+        tagWeight1.m_weight = 1.0f;
+
+        AzFramework::SurfaceData::SurfaceTagWeight tagWeight2;
+        tagWeight2.m_surfaceType = tag2;
+        tagWeight2.m_weight = 0.7f;
+
+        AzFramework::SurfaceData::SurfaceTagWeight tagWeight3;
+        tagWeight3.m_surfaceType = tag3;
+        tagWeight3.m_weight = 0.3f;
+
+        NiceMock<UnitTest::MockTerrainAreaSurfaceRequestBus> mockSurfaceRequests(entity->GetId());
+        ON_CALL(mockSurfaceRequests, GetSurfaceWeights).WillByDefault(
+            [&tagWeight1, &tagWeight2, &tagWeight3](const AZ::Vector3& position, AzFramework::SurfaceData::SurfaceTagWeightList& surfaceWeights)
+            {
+                surfaceWeights.clear();
+                float absYPos = fabsf(position.GetY());
+                if (absYPos < 1.0f)
+                {
+                    surfaceWeights.push_back(tagWeight1);
+                }
+                else if(absYPos < 2.0f)
+                {
+                    surfaceWeights.push_back(tagWeight2);
+                }
+                else
+                {
+                    surfaceWeights.push_back(tagWeight3);
+                }
+            }
+        );
+
+        auto perPositionCallback = [&tagWeight1, &tagWeight2, &tagWeight3](size_t xIndex, size_t yIndex,
+            const AzFramework::SurfaceData::SurfacePoint& surfacePoint, [[maybe_unused]] bool terrainExists)
+        {
+            constexpr float epsilon = 0.0001f;
+            float absYPos = fabsf(surfacePoint.m_position.GetY());
+            if (absYPos < 1.0f)
+            {
+                EXPECT_EQ(surfacePoint.m_surfaceTags[0].m_surfaceType, tagWeight1.m_surfaceType);
+                EXPECT_NEAR(surfacePoint.m_surfaceTags[0].m_weight, tagWeight1.m_weight, epsilon);
+            }
+            else if(absYPos < 2.0f)
+            {
+                EXPECT_EQ(surfacePoint.m_surfaceTags[0].m_surfaceType, tagWeight2.m_surfaceType);
+                EXPECT_NEAR(surfacePoint.m_surfaceTags[0].m_weight, tagWeight2.m_weight, epsilon);
+            }
+            else
+            {
+                EXPECT_EQ(surfacePoint.m_surfaceTags[0].m_surfaceType, tagWeight3.m_surfaceType);
+                EXPECT_NEAR(surfacePoint.m_surfaceTags[0].m_weight, tagWeight3.m_weight, epsilon);
+            }
+        };
+
+        terrainSystem->ProcessSurfaceWeightsFromRegion(testRegionBox, stepSize, perPositionCallback, AzFramework::Terrain::TerrainDataRequests::Sampler::BILINEAR);
+    }
+
+    TEST_F(TerrainSystemTest, TerrainProcessSurfacePointsFromRegion)
```

Comment:
nit: So much duplicate code between these tests. Might be able to factor it out into some kind of testing context that sets up the terrain, surface tags, etc.

Reply:
Yes, in this case, I think I can move the setup of the surface tags to a common function.